apollo-router/src/plugins/telemetry/config_new/selectors.rs

```diff
@@ -3569,4 +3852,203 @@ mod test {
             Some("default".into())
         );
     }
+
+    const TEST_SUBGRAPH_NAME: &str = "test_subgraph_name";
+    const TEST_SOURCE_NAME: &str = "test_source_name";
+    const TEST_URL_TEMPLATE: &str = "/test";
+    const TEST_HEADER_NAME: &str = "test_header_name";
+    const TEST_HEADER_VALUE: &str = "test_header_value";
+    const TEST_STATIC: &str = "test_static";
+
+    #[fixture]
+    fn connector_info() -> ConnectorInfo {
+        ConnectorInfo {
+            subgraph_name: TEST_SUBGRAPH_NAME.to_string(),
+            source_name: Some(TEST_SOURCE_NAME.to_string()),
+            http_method: HTTPMethod::Get.as_str().to_string(),
+            url_template: TEST_URL_TEMPLATE.to_string(),
+        }
+    }
+
+    #[fixture]
+    fn context(connector_info: ConnectorInfo) -> RouterContext {
+        let context = RouterContext::default();
+        context
+            .insert(CONNECTOR_INFO_CONTEXT_KEY, connector_info)
+            .unwrap();
+        context
+    }
+
+    #[fixture]
+    fn http_request(context: RouterContext) -> HttpRequest {
+        HttpRequest {
+            http_request: http::Request::builder().body("".into()).unwrap(),
+            context,
+        }
+    }
```

Comment:
I don't feel confident introducing this technique to debug. We try as much as we can to lower the compilation time and if we introduce macros in our tests it will increase the time. Could you try to use the same kind of tests we did before. If it's something you would like to introduce in the router codebase then I would suggest to talk about this during one of our router arch meeting because we need opinion from the team. Also we have yaml based tests in `apollo-router/src/plugins/telemetry/config_new/fixtures` Could you add more tests directly there please ?

Reply:
> I don't feel confident introducing this technique to debug. We try as much as we can to lower the compilation time and if we introduce macros in our tests it will increase the time. Could you try to use the same kind of tests we did before. If it's something you would like to introduce in the router codebase then I would suggest to talk about this during one of our router arch meeting because we need opinion from the team. Also we have yaml based tests in `apollo-router/src/plugins/telemetry/config_new/fixtures` Could you add more tests directly there please ?

I replaced the use of `rstest` with just plain unit tests, and added more tests in `fixtures`.